lib/elixir/lib/protocol.ex

```diff
@@ -902,6 +902,11 @@ defmodule Protocol do
       @compile {:inline, struct_impl_for: 1}
 
       unless Module.defines_type?(__MODULE__, {:t, 0}) do
+        @typedoc """
+        All terms that implement this protocol.
+
+        This type is mostly useful for documentation.
```

Comment:
I think this verbiage is incorrect. It doesn't return all terms, it returns all types. For example, #PID<0.32.0> is a term, but there is no type that represents it.

Furthermore, this type is refined with protocol consolidation, so it can actually find bugs (but I am not sure if Dialyzer considers it) beyond being for documentation only.

```suggestion
        All types that implement this protocol.
```

Reply:
(I had misread your other suggestions, yes all types makes sense rather than all terms)